src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -81,6 +130,58 @@ def fetch_x509_svid(self) -> X509Svid:
 
         return self._create_x509_svid(svid)
 
+    def watch_x509_context(
+        self,
+        on_success: Callable[[X509Context], None],
+        on_error: Callable[[Exception], None],
+        retry_connect: bool = True,
+    ) -> CancelHandler:
+        """Watches for X.509 context updates.
+
+           This method returns immediately and spawns a new thread to handle the connection with the Workload API. That thread
+           will keep running until the client calls the method `cancel` on the returned CancelHandler, or in case
+           `retry_connect` is false and there is an error returned by the Workload API.
+
+           A new Stream to the Workload API is opened for each call to this method, so that the client starts getting
+           updates immediately after the Stream is ready and doesn't have to wait until the Workload API dispatches
+           the next update based on the SVIDs TTL.
+
+           In case of an error, if `retry_connect` is True and the error was not grpc.StatusCode.CANCELLED
+           or grpc.StatusCode.INVALID_ARGUMENT, it will attempt to establish a new connection
+           to the Workload API, using an exponential backoff policy to perform the retries, starting with a delay of 0.1 seconds,
+           incrementing it then to 0.2, 0.4, 0.8, 1.6 and so on (until the max backoff of 60 seconds). It retries indefinitely.
+
+           # TODO: make the backoff policy configurable
+
+        Args:
+            on_success: A Callable accepting a X509Context as argument and returning None, to be executed when a new update
+                        is fetched from the Workload API,
+
+            on_error: A Callable accepting an Exception as argument and returning None, to be executed when there is
+                      an error on the connection with the Workload API.
+
+            retry_connect: Enable retries when the connection with the Workload API returns an error.
+                           Default: True
+
+        Returns:
+            CancelHandler: An object on which it can be called the method `cancel` to close the stream connection with
+                           the Workload API.
+        """
+
+        cancel_handler = CancelHandler()
+        retry_handler = _RetryHandler()
```

Comment:
_I might be missing something so sorry if it is the case here._
I see that this `retry_handler` is always initialized with and instance of `_RetryHandler()` no matter if `retry_connect` is true or false and then during the execution of this method both vars are passed as argument in several methods. I think that an alternative to avoid passing two arguments could be to initialize  `retry_handler` only when `retry_connect` is `True` (initialized to None in other case) and  then check whether _retry should occurs_ by confirming the presence of a retry handler with `if retry_handler`.